build/build.sh

```diff
@@ -3,12 +3,26 @@
 set -e
 
 OSDIST=`grep '^ID=' /etc/os-release | awk -F= '{print $2}' | tr -d '"'`
+VERSION=`grep '^VERSION_ID=' /etc/os-release | awk -F= '{print $2}' | tr -d '"'`
+MAJOR=${VERSION%.*}
 BUILDDIR=$(readlink -f $(dirname ${BASH_SOURCE[0]}))
 CORE=`grep -c ^processor /proc/cpuinfo`
 CMAKE=cmake
 CMAKE_MAJOR_VERSION=`cmake --version | head -n 1 | awk '{print $3}' |awk -F. '{print $1}'`
 CPU=`uname -m`
 
+# Documentation states that devtoolset-9 is a requirement for building on RHEL/CentOS 7.x
+if [[ ( $OSDIST == "rhel" || $OSDIST == "centos" ) && $MAJOR -lt "8" ]]; then
+    echo "Setting up devtoolset-9"
+    {
+        # Try
+	source scl_source enable devtoolset-9
+    } || {
+        # Except
+	echo "devtoolset-9 is not installed, please run xrtdeps.sh" && exit 1
+    }
+fi
```

Comment:
At the very least this should be conditioned on the version of gcc,  it is possible that a user has installed a more modern version of gcc or modified their env to point to a local install of later gcc.   Enforcing devtoolset-9 is not a good idea..

Personally, I believe the user should set the devtoolset themselves if necessary.  Also 7.x is completely legacy.

Reply:
@stsoe Based on your suggestion, I have modified this logic... now devtoolset-9 is not a requirement... but it is a fallback.

This PR is just for being kind to our existing users who work on RHEL/CentOS 7.x. Which is still a lot of people. For instance the FPGA Developer AMI on AWS is still CentOS 7.5.

Once we drop support for this legacy OS, we can just remove this logic.

The GCCVERSION extraction I added could be useful in the future as well.

Reply:
https://aws.amazon.com/marketplace/pp/prodview-gimv3gqbpe57k